snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy

```diff
@@ -0,0 +1,125 @@
+package scripts
+
+import com.b2international.index.Hits
+import com.b2international.index.query.Expression
+import com.b2international.index.query.Expressions
+import com.b2international.index.query.Query
+import com.b2international.index.revision.RevisionSearcher
+import com.b2international.snowowl.core.ComponentIdentifier
+import com.b2international.snowowl.core.date.EffectiveTimes
+import com.b2international.snowowl.snomed.common.SnomedTerminologyComponentConstants
+import com.b2international.snowowl.snomed.core.domain.constraint.SnomedCardinalityPredicate
+import com.b2international.snowowl.snomed.core.domain.constraint.SnomedConceptSetDefinition
+import com.b2international.snowowl.snomed.core.domain.constraint.SnomedConstraint
+import com.b2international.snowowl.snomed.core.domain.constraint.SnomedRelationshipPredicate
+import com.b2international.snowowl.snomed.core.ecl.EclExpression
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedRelationshipIndexEntry
+import com.b2international.snowowl.snomed.datastore.request.SnomedRelationshipSearchRequestBuilder
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests
+import com.google.common.base.Strings
+import com.google.common.collect.Lists
+import com.google.common.collect.Sets
+
+RevisionSearcher searcher = ctx.service(RevisionSearcher.class)
+List<ComponentIdentifier> issues = Lists.newArrayList()
+def mrcmRules = SnomedRequests.prepareSearchConstraint()
+		.all()
+		.build()
+		.execute(ctx)
+		.stream()
+		.filter({SnomedConstraint constraint -> constraint.getPredicate() instanceof SnomedCardinalityPredicate
+			? ((SnomedCardinalityPredicate) constraint.getPredicate()).getPredicate() instanceof SnomedRelationshipPredicate
+			: constraint.getPredicate() instanceof SnomedRelationshipPredicate})
+		.collect();
+		
+def getApplicableConcepts = { SnomedConceptSetDefinition conceptSetDefinition ->
```

Comment:
Use the actual `ECL` String as function parameter of the `getApplicableConcepts` method.